paderbox/utils/nested.py

```diff
@@ -400,3 +413,196 @@ def squeeze_nested(orig):
         if squeezed and all([orig[key] == orig[keys[0]] for key in keys]):
             return orig[keys[0]]
     return orig
+
+
+def _get_by_path(
+        d: Union[Mapping, Sequence],
+        path: Tuple[Any, ...],
+        broadcast: bool = False
+):
+    """ Helper function for `get_by_path` and `set_by_path`. """
+    for k in path:
+        try:
+            d = d[k]
+        except KeyError:
+            # Not sure if broadcasting makes sense for lists/tuples
+            if broadcast and not isinstance(d, dict):
+                return d
+            raise
+        except TypeError as e:
+            # Not sure if broadcasting makes sense for lists/tuples
+            if broadcast and not isinstance(d, dict):
+                return d
+            raise TypeError(
+                'Only indexable types are supported in `get_by_path` and '
+                '`set_by_path`!',
+                d, k
+            ) from e
+    return d
+
+
+def get_by_path(
+        d: Union[Mapping, Sequence],
+        path: Union[str, Tuple[Any, ...], None],
+        *,
+        broadcast: bool = False,
```

Comment:
I don't like the name `broadcast`. But I have no better name, so it is probably fine.